### PR TITLE
#45 Remove StorageAddress from Availability

### PR DESCRIPTION
**Pull request by a-v-dubr**

### CODE REVIEW

File: Infrastructure/Helper/DbSchemaCheck.cs [hunk 1]
```diff
@@ -73,5 +74,40 @@ private static void EnsurePaidCashColumnExistsInSalesTable(SQLiteConnection conn
...
+                using (var cmd = new SQLiteCommand("", connection, transaction))
+                {
+                    cmd.CommandText = $"SELECT COUNT(*) AS ColumnExists FROM sqlite_master WHERE type = 'table' AND name = 'Avaliability' AND sql LIKE '%StorageAdress%';";
+                    try
+                    {
+                        bool columnExists = Convert.ToInt32(cmd.ExecuteScalar()) != 0;
+                        if (columnExists)
+                        {
+                            cmd.Parameters.Clear();
+                            cmd.CommandText = $"CREATE TABLE Avaliability_backup " +
```
Comment (dmytroshchotkin) [inline Infrastructure/Helper/DbSchemaCheck.cs]: Why do you need to recreate the table? Can't you just remove the column, what's the problem?

File: Infrastructure/Helper/DbSchemaCheck.cs [hunk 2]
```diff
@@ -73,5 +74,40 @@ private static void EnsurePaidCashColumnExistsInSalesTable(SQLiteConnection conn
                 }
             }
         }
+
+        private static void EnsureStorageAddressColumnDeletedFromAvailabilityTable(SQLiteConnection connection)
+        {
+            using (var transaction = connection.BeginTransaction())
```
Comment (dmytroshchotkin) [inline Infrastructure/Helper/DbSchemaCheck.cs]: Do you really need a transaction here?

**Review by dmytroshchotkin — CHANGES_REQUESTED** [review]

File: Infrastructure/Helper/DbSchemaCheck.cs (hunk 1 above)
Comment (a-v-dubr) [reply Infrastructure/Helper/DbSchemaCheck.cs]: 'drop column' query doesn't work obviously in SQLite; 
removed any StorageAddress mentions from DB operations and from Purchase UI, added 2nd goal in User Story #45 (drop SA column from DB table without recreating Availability table)

File: Infrastructure/Helper/DbSchemaCheck.cs (hunk 2 above)
Comment (a-v-dubr) [reply Infrastructure/Helper/DbSchemaCheck.cs]: yes, because there are used two commands 
image

**Review by dmytroshchotkin — APPROVED** [review]

**Pull request merged**